kythe/typescript/indexer.ts

```diff
@@ -32,15 +32,47 @@ export interface VName {
   language: string;
 }
 
+/**
+ * A indexer host holds information about the program indexing and methods used
+ * by the TypeScript indexer that may also be useful to plugins, reducing code
+ * duplication.
+ */
+export interface IndexerHost {
```

Comment:
"Host" just doesn't sit well. What about having this be "IndexerContext" and the implementation something (java-esque, I know) like "StandardIndexerContext"?

Reply:
Works for me,

Reply:
My suggestion of "Host" here is intentional, because it's what the TypeScript compiler uses.  For example as users of it we provide a CompilerHost, ModuleResolutionHost, FormatDiagnosticsHost and so on.  Looking at the TS API it looks like they define ~12 "Host" interfaces.

Reply:
Ah. As unintuitive as it reads (to me), aligning with the compiler makes sense.

Reply:
Moved the interface to `IndexerHost`. The class is still `StandardIndexerContext`. Let me know if either of you think that should be renamed too :)